src-built-in/components/advancedAppLauncher/src/stores/StoreActions.js

```diff
@@ -449,38 +498,58 @@ function deleteFolder(folderName) {
 		_setValue("activeFolder", data.activeFolder);
 	}
 
+	const deletedFolders = data.deleted;
+	deletedFolders.push(folderName);
+
 	delete data.folders[folderName] && _setFolders(() => {
 		// Update the order of folders
 		const index = data.foldersList.indexOf(folderName);
 		data.foldersList.splice(index, 1);
 		_setValue("appFolders.list", data.foldersList);
+		_setValue("deleted", deletedFolders);
 	});
 }
 
 function renameFolder(oldName, newName) {
 	let oldFolder = data.folders[oldName];
 	data.folders[newName] = oldFolder;
+	delete data.folders[oldName];
 	_setFolders(() => {
 		let indexOfOld = data.foldersList.findIndex((folderName) => {
 			return folderName === oldName;
 		});
 		data.foldersList[indexOfOld] = newName;
+
+		// If the name the user is attempting to rename to is the name of an old deleted folder
+		// remove the key from deleted and allow rename
+		if (data.deleted.includes(newName)) {
+			const index = data.deleted.indexOf(newName);
+			const deletedFolders = data.deleted;
+			deletedFolders.splice(index, 1);
+			_setValue("deleted", deletedFolders);
+		}
+
 		_setValue("appFolders.list", data.foldersList);
 		delete data.folders[oldName];
 	});
 }
 
 function addAppToFolder(folderName, app) {
-	data.folders[folderName].apps[app.appID] = {
-		name: app.name,
-		displayName: app.displayName,
-		appID: app.appID
-	};
-	_setFolders();
+	const appIndex = _findAppIndexInFolders(app.appID, folderName);
+
+	if (appIndex < 0) {
```

Comment:
This is a slight change in behavior as it appears we could overwrite file information if we called this on an app that was already in the folder. Hopefully we're not relying on that case as this behavior seems more intuitive.

Reply:
That's why it checks that appIndex is less than 0. If appIndex is >= 0, the app already exists and won't be overwritten. Perhaps that is its _own_ issue, but either way it has to work one way or the other. Either we allow overwrite or we prevent it

Reply:
Yup, I wanted to note the change in behavior, but I think your way is better. I think we allowed it to be overwritten before, but that doesn't mean we actually used this function that way.